src/renderer/html_handlebars/hbs_renderer.rs

```diff
@@ -209,3 +214,37 @@ fn make_data(book: &MDBook) -> Result<serde_json::Map<String, serde_json::Value>
     debug!("[*]: JSON constructed");
     Ok(data)
 }
+
+fn build_header_links(mut html: String) -> String {
+    for header in &["h1", "h2", "h3", "h4", "h5"] {
+        let regex = Regex::new(&format!("<{h}>(.*?)</{h}>", h=header)).unwrap();
+
+        html = regex.replace_all(&html, |caps: &Captures| {
+            let text = &caps[1];
+            let mut id = text.to_string();
+            let repl_sub = vec!["<em>", "</em>", "<code>", "</code>",
+                                "<strong>", "</strong>",
+                                "&lt;", "&gt;", "&amp;", "&#39;", "&quot;"];
```

Comment:
i copied this code directly from rustdoc so that it'd have the same behavior; i agree that it's probably not exhaustive.

> I am not sure how GitHub does it.


I don't know either :/